src/trace/context.ts

```diff
@@ -108,6 +151,14 @@ export function extractTraceContext(
       }
     }
     return trace;
+  } else {
+    // do not send step functions context to xray
+    if (stepFuncContext) {
```

Comment:
nitpik: can we combine this with the block above in order to put the `stepFuncContext` related logic together. that is:
```javascript
  if (stepFuncContext) {
    try {
      addStepFunctionContextToXray(stepFuncContext);
    } catch (error) {
      if (error instanceof Error) {
        logError("couldn't add step function metadata to xray", error as Error);
      }
    }
    if (!trace) {
      trace = readTraceFromStepFunctionsContext(stepFuncContext);
      if (trace !== undefined) {
        return trace;
      }
    }
  }
```

Reply:
Great suggestion. Updated and thanks for the review!